### ISSUE TITLE
CI: Etat du Stock - add map and table

### ISSUE THREAD

**Issue by mark-prins** [msupply-foundation/msupply-dashboard#54]
Request from the Feb 2020 dashboard review.

Add a map at the top which shows nothing until a program and item is selected.
The map is interactive, when you click a district then a table is shown

District | Out of Stock | Overstock | Normal | Understock
-- | -- | -- | -- | --
  | #of stores with this status |   |   |  

Note: They wrote column headers of R,O,N,U - for now use the words as they have more meaning.
(Retour for the out of stock, hence the 'R')

District are colour coded based on the status: more detail to come on this one.

Include a legend.

There are percentages to determine the status of the district, based on stores in that district
eg. If, for a district, 
% of stores at ‘normal’ is < 15% then district is out
% of stores at ‘normal’ is < 30% then district is under
% of stores at ‘normal’ is < 75% then district is normal
else district is over

**Comment by CarlosNZ** [comment]
Just wondering which of the existing Dashboard pages this should be integrated into?

**Comment by mark-prins** [comment]
I think make a new dashboard for it. There was much discussion in the UAT session about this one which makes me think that it's a big enough deal to warrant its own space.

**Comment by CarlosNZ** [comment]
I've implemented a initial version of this here: https://ci.msupply.org:3000/d/jMp9xTqWz/map-and-table?orgId=1

It's a bit tricky, because so many items are not present at so many stores and districts that a lot of the filter choices just result in an empty map or table. Takes a while to find a combo that shows _anything_ meaningful, hence I've saved it with a variable setting that at least shows _something_.

@mark-prins -- do you know any way to write queries that are conditional on the state of the grafana variable? I mean, like, we would normally have a query like:
`SELECT * FROM some_table WHERE value = $grafana_variable`
but is it at all possible to do the equivalent of:
`SELECT * FROM some_table WHERE $grafana_variable = <some_value>`
Basically, asking if there's way to pass a variable as a field name rather than just a value of a field?

I need to make the map regions click-able (hopefully it doesn't take too long to remember how it worked for the circles in the other map panel), and I'll try and add the customisable tool-tip field like you did in the other map, too.

Any other suggestions to make this closer to the requirements?

**Comment by mark-prins** [comment]
I think that looks great! I've added a legend - in order to try out the text component and see if it was possible.

One small bug is that selecting `All` *Items* breaks the map. Query just needs `in` rather than `=`

I think the data will improve as we roll out more stores, though it is a concern that so few combinations show results

**Comment by CarlosNZ** [comment]
> One small bug is that selecting `All` _Items_ breaks the map. Query just needs `in` rather than `=`

Right -- done! :)
